Cr daemon

boucher opened the pull request.

Commit:
Checkpoint/Restore Support: add exec driver methods

Methods for checkpointing and restoring containers were added to the
native driver.  The LXC driver returns an error message that these
methods are not implemented yet.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	daemon/execdriver/native/create.go
	daemon/execdriver/native/driver.go
	daemon/execdriver/native/init.go

Commit:
Checkpoint/Restore Support: add functionality to daemon

Support was added to the daemon to use the Checkpoint and Restore methods
of the native exec driver for checkpointing and restoring containers.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	api/server/server.go
	daemon/container.go
	daemon/daemon.go
	daemon/networkdriver/bridge/driver.go
	daemon/state.go
	vendor/src/github.com/docker/libnetwork/ipallocator/allocator.go

huikang committed:
Release the network resource during checkpoint

Restore failed if network resource not released during checkpoint,
e.g., a container with port open with -p

Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

Conflicts:
	daemon/container.go

boucher committed:
Update checkpoint/restore support to match docker/master

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

huikang commented:
@boucher 
When I compile your cr-daemon branch, it gives me the 
These files are not properly gofmt'd:
- daemon/checkpoint.go
- daemon/container.go
- daemon/state.go
- runconfig/restore.go

Please reformat the above files using "gofmt -s -w" and commit t                 he result.

I can bypass this issue by comment out some line in hack/make.sh. However you may consider reformat it. My gofmt setup has some problem and could not resolve all the format error.
- Hui

boucher commented:
@huikang Should be resolved now

boucher committed:
Update checkpoint/restore support in the daemon.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)
